### ISSUE TITLE
Async extendDefaultFields support?

### ISSUE THREAD

**Issue by zypA13510** [mweibel/connect-session-sequelize#72]
I'm trying to do what is described on the wiki - adding a userId field to my session model. However, I'm also trying to constrain the userId field with an FK. So for a new user, I need to create the entry in users first. I'm going to do this in the extendDefaultFields, but turns out it does not support async function.

I've taken a look at the code, [connect-session-sequelize.js#L87](https://github.com/mweibel/connect-session-sequelize/blob/master/lib/connect-session-sequelize.js#L87), it should have no problem supporting async function, since we are going to call `fn` back anyway.

**Comment by mweibel** [comment]
So what is your question? Can you post example code to be able to reproduce it?

**Comment by zypA13510** [comment]
The question is, connect-session-sequelize does not support extendDefaultFields method returning a Promise.
```Javascript
extendDefaultFields: async (defaults, data) => {
	var sUserId = data && data.passport && data.passport.user && data.passport.user.id;
	if (sUserId) {
		await User.findOrCreate({
			where: {
				userId: sUserId
			},
			defaults: {
				firstName: data.passport.user.first_name,
				lastName: data.passport.user.last_name,
				email: data.passport.user.email,
			},
		});
		return {
			data: defaults.data,
			expires: defaults.expires,
			userId: sUserId,
		};
	} else {
		return {
			data: defaults.data,
			expires: defaults.expires,
		};
	}
}
```

**Comment by zypA13510** [comment]
Just tried this [zypA13510/connect-session-sequelize](https://github.com/zypA13510/connect-session-sequelize), works fine for me. But I'm not sure if it will break existing codes or on older environment.

**Comment by zypA13510** [comment]
@mweibel Yes, the userId comes from an external system. The middleware provided by the SSO provider would directly interact with `express-session`. So I have to find a place somewhere in `express-session` or in the session store (`connect-session-sequelize`) to create the user if it doesn't exist.
Of course, I also feel that this is probably not the "correct" place to add this code, but it's the best option I find. To support this or not is your decision though, personally, I feel that it does no harm anyway (because the existing extendDefaultFields method does not require synchronous behavior, the worst scenario is probably a delay in the resolution of the Promise returned).

**Comment by zypA13510** [comment]
Hi @mweibel,

Just want to get back to this thread to discuss another scenario I've encountered - you may want to perform an asynchronous operation during `extendDefaultFields` e.g. perform a lookup using sequelize, which would be common I believe, since everyone who uses this package would be using sequelize, and sequelize methods are mostly asynchronous.

So I think maybe you should reconsider this. (Not necessarily using async/await because this feature is only supported by Node.js v7.6 and above, but you can use callback routine)
